### PR TITLE
VideoCommon: Fix some warnings.

### PR DESCRIPTION
**Pull request by phire**
This change is [Reviewable](https://reviewable.io/reviews/dolphin-emu/dolphin/3950)

### CODE REVIEW

File: Source/Core/VideoCommon/LightingShaderGen.cpp [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+
+// vertex shader
+// lights/colors
+// materials name is I_MATERIALS in vs and I_PMATERIALS in ps
+// inColorName is color in vs and colors_ in ps
+// dest is o.colors_ in vs and colors_ in ps
+void GenerateLightingShaderCode(ShaderCode& object, const LightingUidData& uid_data,
+                                       int components, const char* inColorName, const char* dest)
+{
+  for (unsigned int j = 0; j < xfmem.numChan.numColorChans; j++)
```
Comment (phire) [inline Source/Core/VideoCommon/LightingShaderGen.cpp]: TODO: shouldn't be reading from xfmem here

(to be fixed in another PR)

**Comment by degasus** [comment]
:lgtm:

---

Reviewed 7 of 7 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3950#-:-KLBUintdoLcjmpA1SEF:1417088721)*
